bump.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+# ----------- Basic Checking -----------
+
+# Check that 1 arg has been supplied
+if [ "$#" != "1" ]; then
+	echo "[-] Exactly one argument is needed."
+	echo '[?] Example: `./bump.sh 0.2.6`'
+	exit 1
+fi
+
+# Check that required binaries are installed
+for package in semver git npm; do
+	if [ ! -f "/usr/bin/$package" ]; then
+		echo "[-] `$package` is needed to bump version"
+		exit 1
+	fi
+done
+
+# Check that supplied version is semantically correct
+if [ ! "$1" = "$(/usr/bin/semver $1)" ]; then
+	echo "[-] $1 is not a valid version number according to semver"
+	exit 1
+fi
+
+# Check that git working directory is clean...
+if [ -n "$(git status -s --untracked-files=no)" ]; then
+	echo "[-] git working directory isn't clean"
+	exit 1
+fi
+
+# ----------- Variable definition -----------
+
+# Define needed variables
+CURRENT=$(npm pkg get version | sed 's/"//g')
+NEW_VERSION=$1
+BUMP_BRANCH="${NEW_VERSION}-version-bump"
+ISCSC_REMOTE=$(git remote -v | grep 'git@github.com:iScsc/iscsc.fr.git' | awk '{print $1}' | head --lines 1)
+
+# ----------- Version checking -----------
+
+# Check if targeted version is > than current
+echo "[+] Current version is '${CURRENT}'"
+if [ ! $(semver "${NEW_VERSION}" -r ">$CURRENT") ]; then
+	echo "[-] '${NEW_VERSION}'<='$CURRENT', '${NEW_VERSION}' isn't accepted as new version."
+	exit 1
+fi
+echo "[+] '${NEW_VERSION}'>'$CURRENT', '${NEW_VERSION}' is accepted as new version."
+
+# ----------- Git setup -----------
+
+# ...and checkout on main to create a version bump branch
+echo "[+] Checkout on ${ISCSC_REMOTE}/main"
+[ -z "$DRY_RUN" ] && git checkout ${ISCSC_REMOTE}/main
+echo "[+] switching to ${BUMP_BRANCH}"
+[ -z "$DRY_RUN" ] && git switch -c ${BUMP_BRANCH}
+
+# ----------- Version Bump -----------
+
+echo '[+] Bumping `frontend`'
+cd frontend
+[ -z "$DRY_RUN" ] && npm version "${NEW_VERSION}" --no-git-tag-version
```

Comment:
What's `--no-git-tag-version`?

Reply:
By default `npm version` will commit and tag the change [see the doc](https://docs.npmjs.com/cli/v8/commands/npm-version#git-tag-version).
this flag is used to disable this behavior
[doc](https://docs.npmjs.com/cli/v8/commands/npm-version#description)
```
If run in a git repo, it will also create a version commit and tag. This behavior is controlled by git-tag-version (see below), and can be disabled on the command line by running npm --no-git-tag-version version. It will fail if the working directory is not clean, unless the -f or --force flag is set.
```